File: .skills/openclaw-skills/skills/vnagin/zoom-manager-clawd/scripts/get_meeting_info.js
#!/usr/bin/env node
const fs = require('fs').promises;
const path = require('path');
const fetch = require('node-fetch');

// Load config
async function loadConfig() {
  const configPath = path.join(__dirname, '..', 'config.json');
  const data = await fs.readFile(configPath, 'utf8');
  return JSON.parse(data);
}

// Get OAuth access token
async function getAccessToken(config) {
  const url = 'https://api.zoom.us/oauth/token';
  const params = new URLSearchParams({
    grant_type: 'account_credentials',
    account_id: config.account_id
  }).toString();

  const response = await fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded',
      'Authorization': `Basic ${Buffer.from(`${config.client_id}:${config.client_secret}`).toString('base64')}`
    },
    body: params
  });

  if (!response.ok) {
    throw new Error(`Failed to get access token: ${response.status} ${response.statusText}`);
  }

  const data = await response.json();
  return data.access_token;
}

// Get meeting info
async function getMeetingInfo(meetingId, accessToken) {
  const url = `https://api-us.zoom.us/v2/meetings/${meetingId}`;
  const response = await fetch(url, {
    method: 'GET',
    headers: {
      'Authorization': `Bearer ${accessToken}`
    }
  });

  if (!response.ok) {
    throw new Error(`Failed to get meeting info: ${response.status} ${response.statusText}`);
  }

  return await response.json();
}

// Main execution
async function main() {
  if (process.argv.length !== 3) {
    console.error('Usage: node get_meeting_info.js <meeting_id>');
    process.exit(1);
  }

  const meetingId = process.argv[2];
  try {
    const config = await loadConfig();
    const token = await getAccessToken(config);
    const info = await getMeetingInfo(meetingId, token);
    console.log(JSON.stringify(info, null, 2));
  } catch (error) {
    console.error('Error:', error.message);
    process.exit(1);
  }
}

main();

File: .skills/openclaw-skills/skills/vnagin/zoom-manager-clawd/scripts/get_meeting_info.py
#!/usr/bin/env python3
"""
Get detailed information about a specific Zoom meeting.
Usage: python get_meeting_info.py <meeting_id>
"""

import os
import sys
import json
import requests

# Load config
CONFIG_PATH = os.path.join(os.path.dirname(__file__), '..', 'config.json')
with open(CONFIG_PATH, 'r') as f:
    config = json.load(f)

ACCOUNT_ID = config['account_id']
CLIENT_ID = config['client_id']
CLIENT_SECRET = config['client_secret']
USER_ID = config['user_id']

def get_access_token():
    """Get OAuth access token."""
    url = "https://api.zoom.us/oauth/token"
    data = {
        "grant_type": "account_credentials",
        "account_id": ACCOUNT_ID
    }
    response = requests.post(url, auth=(CLIENT_ID, CLIENT_SECRET), data=data)
    response.raise_for_status()
    return response.json()['access_token']

def get_meeting_info(meeting_id):
    """Get details of a specific meeting."""
    token = get_access_token()
    url = f"https://api-us.zoom.us/v2/meetings/{meeting_id}"
    headers = {
        "Authorization": f"Bearer {token}"
    }
    response = requests.get(url, headers=headers)
    response.raise_for_status()
    return response.json()

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python get_meeting_info.py <meeting_id>")
        sys.exit(1)

    meeting_id = sys.argv[1]
    try:
        info = get_meeting_info(meeting_id)
        print(json.dumps(info, indent=2))
    except requests.exceptions.HTTPError as e:
        print(f"Error: {e}")
        print(f"Response: {e.response.text}")
    except Exception as e:
        print(f"Unexpected error: {e}")